core/src/main/python/wlsdeploy/tool/create/domain_creator.py

```diff
@@ -276,54 +276,66 @@ def __run_rcu(self):
 
             rcu_runner_map = dict()
-            atp_conn_properties = {}
-
-            # update password fields with decrypted passwords
-            if rcu_db_info.get_keystore_password() is not None:
-                atp_conn_properties[DRIVER_PARAMS_KEYSTOREPWD_PROPERTY] \
-                    = {'Value': rcu_db_info.get_keystore_password()}
-
-            if rcu_db_info.get_truststore_password() is not None:
-                atp_conn_properties[DRIVER_PARAMS_TRUSTSTOREPWD_PROPERTY] \
-                    = {'Value': rcu_db_info.get_truststore_password()}
-
-            atp_conn_properties[DRIVER_PARAMS_NET_TNS_ADMIN] = { 'Value': rcu_db_info.get_tns_admin()}
-            atp_conn_properties[DRIVER_PARAMS_NET_SSL_VERSION] = { 'Value': 1.2 }
-            atp_conn_properties[DRIVER_PARAMS_NET_FAN_ENABLED] = { 'Value': 'false' }
-            atp_conn_properties[DRIVER_PARAMS_NET_SERVER_DN_MATCH_PROPERTY] = { 'Value': 'false' }
-            atp_conn_properties[DRIVER_PARAMS_TRUSTSTORETYPE_PROPERTY] = { 'Value': 'JKS' }
-            atp_conn_properties[DRIVER_PARAMS_KEYSTORETYPE_PROPERTY] = { 'Value': 'JKS' }
-            atp_conn_properties[DRIVER_PARAMS_TRUSTSTORE_PROPERTY] = { 'Value': rcu_db_info.get_tns_admin()
-                                                                                + os.sep + "truststore.jks" }
-            atp_conn_properties[DRIVER_PARAMS_KEYSTORE_PROPERTY] = { 'Value': rcu_db_info.get_tns_admin()
-                                                                        + os.sep + "keystore.jks"}
-
-            if not atp_conn_properties.has_key(DRIVER_PARAMS_NET_FAN_ENABLED):
-                atp_conn_properties[DRIVER_PARAMS_NET_FAN_ENABLED] = { 'Value' : 'false'}
+            ssl_conn_properties = dict()
+
+            # historical reason, these values may not be there, and assume default name for now, user can override with
+            # sso file name
+
+            truststore = rcu_db_info.get_truststore()
+            keystore = rcu_db_info.get_keystore()
+            truststore_type = rcu_db_info.get_truststore_type()
+            keystore_type = rcu_db_info.get_keystore_type()
+            keystore, keystore_type, truststore, truststore_type = atp_helper.fix_store_type_and_default_value(keystore,
+                                                                        keystore_type, truststore, truststore_type)
+
+
+            self._set_rcu_ssl_args_properties(ssl_conn_properties, rcu_db_info, keystore, keystore_type, truststore,
+                                              truststore_type)
+
+            # hard coding for now, may need to expose it if ATP access changed later
+
+            if not ssl_conn_properties.has_key(DRIVER_PARAMS_NET_FAN_ENABLED):
+                ssl_conn_properties[DRIVER_PARAMS_NET_FAN_ENABLED] = 'false'
+            ssl_conn_properties[DRIVER_PARAMS_NET_SSL_VERSION] = '1.2'
```

Comment:
why don't we set this only if it is not set, like you are doing for the previous property?

Reply:
Actually, the fan enabled should always be false, plus these properties must be that value for it to work properly.    There is comment in the code, not sure why it didn't show up in PR.